packages/oats/src/generate-types.ts

```diff
@@ -575,10 +560,24 @@ export function run(options: Options) {
       ],
       ts.createIdentifier('reflection'),
       undefined,
-      ts.createTypeReferenceNode(fromLib('reflection', 'NamedTypeDefinition'), [
+      ts.createTypeReferenceNode(fromLib('reflection', 'NamedTypeDefinitionDeferred'), [
```

Comment:
this is `() => NamedTypeDefinition`

Reply:
What if we generated 
```ts
public static reflection = () => { return typeNonNullableNullable; };`
```

instead of:
```ts
public static reflection: oar.reflection.NamedTypeDefinitionDeferred<NonNullableNullable> = () => { return typeNonNullableNullable; };
```

The type definition there seems redundant since `typeNonNullableNullable` will always be typed correctly. Normally I wouldn't care, but if we're trying to reduce file size 🤷

Reply:
the type should get erased when building to js so it should not matter. OTOH doesn't ts require exported functions to be explicitly typed?